### ISSUE TITLE
Differing Theme-UI versions breaking in gatbsy-starter-notes-theme

### ISSUE THREAD

**Issue by lannonbr** [gatsbyjs/gatsby#24888]
## Description

Gatsby Admin is using an alpha version of `theme-ui`, at the time of writing this, `0.4.0-alpha-3`. This is causing issues given `gatsby-plugin-theme-ui` uses `0.2.52`.

as of 0.3.x of theme-ui the `ColorMode` component was removed from a named export from the root `theme-ui` package. `gatsby-plugin-theme-ui` 0.2.x uses it here: https://github.com/system-ui/theme-ui/blob/v0.2.53/packages/gatsby-plugin-theme-ui/src/provider.js

This is causing issues with npm's module resolution. if you install with yarn it works fine.

cc @mxstbr @johno @jxnblk

### Steps to reproduce

1. Clone down [gatsby-starter-notes-theme](https://github.com/gatsbyjs/gatsby-starter-notes-theme)
1. Run `npm install`
1. Run `gatsby develop`
1. See the error in the browser of `Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got undefined. ...`
1. Check `node_modules/theme-ui` and confirm that its package.json has a version number of `0.4.x`.

### Expected result

each package resolves properly. ones that depend on `0.2.x` or `0.4.x` use each properly.

### Actual result

NPM's module resolution had`^0.2.x` go to `0.4.x` and broke due to a semver misuse.

### Environment

```
  System:
    OS: macOS Mojave 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.16.3 - ~/.nvm/versions/node/v12.16.3/bin/node
    Yarn: 1.15.2 - ~/.yarn/bin/yarn
    npm: 6.14.4 - ~/.nvm/versions/node/v12.16.3/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 83.0.4103.97
    Edge: 83.0.478.45
    Firefox: 76.0.1
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.22.15 => 2.22.15 
    gatsby-theme-notes: ^1.3.57 => 1.3.57 
```

**Comment by mxstbr** [comment]
Kinda a duplicate of #24653, likely fixed by #24806 or #24810!

**Comment by lannonbr** [comment]
@mxstbr This still hasn't been resolved even though #24810 was merged in given Lerna seems to only updated the yarn.lock file and not the package-lock.json file.

**Comment by mxstbr** [comment]
I'm not sure what that means. Why would the yarn.lock or package-lock.json files be relevant here? The dependencies shouldn't be installed at all 🤔
